lib/device_api/android/adb.rb

```diff
@@ -236,10 +236,11 @@ def self.monkey(qualifier, args)
       def self.screencap( qualifier, args )
         
         filename = args[:filename] or raise "filename not provided (:filename => '/tmp/myfile.png')"
-        
+ 
         convert_carriage_returns = %q{perl -pe 's/\x0D\x0A/\x0A/g'}
         cmd = "screencap -p | #{convert_carriage_returns} > #{filename}"
         
+        cmd = "screencap -p" if getprop(qualifier)['ro.build.version.release'].to_i > 6
```

Comment:
This doesn't appear to write out to `filename`. Also, this and the previous line can be simplified either with an if-then-else or by concatenating the string when version <= 6 instead of setting twice.